admin/create-theme/theme-media.php

```diff
@@ -18,105 +19,55 @@ public static function get_media_folder_path_from_url( $url ) {
 		return $folder_path;
 	}
 
-	public static function get_media_absolute_urls_from_blocks( $flatten_blocks ) {
+	/**
+	 * Get the absolute URLs of all media files for a template
+	 */
+	public static function get_media_absolute_urls_from_template( $template ) {
+
+		$template_blocks = parse_blocks( $template->content );
+		$blocks          = _flatten_blocks( $template_blocks );
```

Comment:
Just a note:
same parsing is being done while trying to remove site specific attributes [here](https://github.com/WordPress/create-block-theme/pull/524/files). All of it can be done in a single place to avoid redundancy of the heavy operations. I am going to iterate on the other PR once this is merged.

Reply:
I don't think this particular function modifies anything (site specific or not), it's just gathering up all of the links to media so that the location of the media can be transitioned.

I do agree though that we should be cleaning out all the site-specific stuff in a single swoop to keep the processing down.